src/systems.rs

```diff
@@ -112,30 +102,42 @@ impl DownloadRateLimiter {
 }
 
 #[derive(Resource)]
-pub struct ActiveDownloads(Arc<AtomicUsize>);
+pub struct DownloadSemaphore(Arc<Semaphore>);
+
+impl DownloadSemaphore {
+    fn new(n: usize) -> Self {
+        Self(Arc::new(Semaphore::new(n)))
+    }
 
-impl Default for ActiveDownloads {
-    fn default() -> Self {
-        Self(Arc::new(AtomicUsize::new(0)))
+    fn semaphore(&self) -> Arc<Semaphore> {
+        Arc::clone(&self.0)
     }
 }
 
+pub(crate) fn initialize_semaphore(
```

Comment:
[MEMO] This system is only needed to initialize the `Semaphore` using the value set in the `SlippyTilesSettings` resource.

Another option is to move `max_concurrent_downloads` from `SlippyTilesSettings` into `SlippyTilesPlugin` and initialize the `Semaphore` in `SlippyTilesPlugin::build`. Personally, I prefer this approach for initializing plugins 😊